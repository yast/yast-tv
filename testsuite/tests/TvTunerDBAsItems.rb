# encoding: utf-8

module Yast
  class TvTunerDBAsItemsClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      Tv.tuners_database = nil
      @items = Tv.TunersDBAsItems(nil, nil)
      DUMP(@items)


      Tv.tuners_database = {}
      @items = Tv.TunersDBAsItems("bttv", nil)
      DUMP(@items)

      Tv.tuners_database = {
        "bttv" => [
          { "name" => "No tuner", "parameters" => { "tuner" => "4" } },
          { "name" => "Alps HSBH1", "parameters" => { "tuner" => "9" } }
        ]
      }
      # choosing bad tuner
      @items = Tv.TunersDBAsItems(
        "bttv",
        { "name" => "Alps HSBH2", "parameters" => { "tuner" => "15" } }
      )
      DUMP(@items)

      # choosing tuner from database
      @items = Tv.TunersDBAsItems(
        "bttv",
        { "name" => "Alps HSBH1", "parameters" => { "tuner" => "9" } }
      )
      DUMP(@items)

      nil
    end
  end
end

Yast::TvTunerDBAsItemsClient.new.main
