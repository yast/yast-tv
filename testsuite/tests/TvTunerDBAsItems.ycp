{

    include "testsuite.ycp";

    map READ = $[
	"etc" : $[ "install_inf" : $[] ],
	"target" : $[ "ycp" : $[] ],
	"probe" : $[ "system" : []],
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

	Tv::tuners_database = nil;
	list items = Tv::TunersDBAsItems(nil, nil);
	DUMP(items);


	Tv::tuners_database = $[];
	items = Tv::TunersDBAsItems("bttv", nil);
	DUMP(items);

	Tv::tuners_database = $["bttv" :
		[ $["name":"No tuner",
			"parameters":$["tuner":"4"]],
		  $["name":"Alps HSBH1",
		    "parameters":$["tuner":"9"]]]
	];
	// choosing bad tuner
	items = Tv::TunersDBAsItems("bttv",$["name":"Alps HSBH2",
										 "parameters":$["tuner":"15"]]);
	DUMP(items);

	// choosing tuner from database
	items = Tv::TunersDBAsItems("bttv",$["name":"Alps HSBH1",
										 "parameters":$["tuner":"9"]]);
	DUMP(items);

}
