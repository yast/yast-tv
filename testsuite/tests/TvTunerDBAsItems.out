Read	.target.size "/etc/install.inf" -1
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system nil
Dump	[]
Dump	[`item (`id (0), "Default (detected)", true)]
Dump	[`item (`id (0), "Default (detected)", false), `item (`id (1), "No tuner", false), `item (`id (2), "Alps HSBH1", false)]
Dump	[`item (`id (0), "Default (detected)", false), `item (`id (1), "No tuner", false), `item (`id (2), "Alps HSBH1", true)]
